Library/Homebrew/tab.rb

```diff
@@ -242,6 +242,26 @@ def compiler
     super || DevelopmentTools.default_compiler
   end
 
+  def homebrew_tag
+    homebrew_version.sub(/\-\d+\-g([a-f0-9]+)(?:\-dirty)?\Z/, "")
```

Comment:
Does it fail to just do a direct version comparison on this value without modification?